lib/sqlalchemy/sql/elements.py

```diff
@@ -123,11 +124,38 @@
 _NMT = TypeVar("_NMT", bound="_NUMBER")
 
 
+@overload
 def literal(
     value: Any,
-    type_: Optional[_TypeEngineArgument[_T]] = None,
+    type_: _TypeEngineArgument[_T],
     literal_execute: bool = False,
 ) -> BindParameter[_T]:
+    ...
+
+
+@overload
+def literal(
```

Comment:
maybe it should infer the type from the value here? sqlalchemy seems to do this at runtime, since this
```py
foo = sa.literal(123)
print(foo.type)
```
prints
```
INTEGER
```

Reply:
@zzzeek opinions here?